SETUP.md

```diff
@@ -0,0 +1,154 @@
+# Sustineo Setup Guide
+
+Sustineo is an application that combines AI capabilities with real-time interactions for voice and image generation.
+
+## Prerequisites
+
+- [Python 3.11](https://www.python.org/downloads/) or later for the API
+- [Node.js 20](https://nodejs.org/) or later for the web component
+- Azure subscription with the following services:
+  - Azure OpenAI service (for voice capabilities)
+  - Azure OpenAI service (for image generation)
+  - Azure Cosmos DB account
+  - Azure Blob Storage account
+
+## Environment Setup
+
+### API Environment Variables
+
+Create a `.env` file in the root directory with the following variables:
+
+```
+# Azure OpenAI for Voice
+AZURE_VOICE_ENDPOINT=https://your-voice-openai-resource.openai.azure.com
+AZURE_VOICE_KEY=your-voice-openai-key
+
+# Azure OpenAI for Image Generation
+AZURE_IMAGE_ENDPOINT=https://your-image-openai-resource.openai.azure.com
+AZURE_IMAGE_API_KEY=your-image-openai-key
+
+# Azure Storage
+SUSTINEO_STORAGE=https://yourstorageaccount.blob.core.windows.net
+
+# Azure Cosmos DB
+COSMOSDB_CONNECTION=your-cosmosdb-connection-string
+
+# Optional: Azure Foundry (if using)
+FOUNDRY_CONNECTION=your-foundry-connection-string
+
+# Optional: Local tracing
+LOCAL_TRACING_ENABLED=false
+```
+
+### Web Environment Setup
+
+The web component uses the API endpoints defined in `web/store/endpoint.ts`. By default, these are set to:
+- WebSocket Endpoint: `ws://localhost:8000`
+- API Endpoint: `http://localhost:8000`
+- Web Endpoint: `http://localhost:5173`
+
+Modify these values if you're deploying to a different environment.
+
+## Installation
+
+### API Setup
+
+1. Navigate to the API directory:
+   ```bash
+   cd api
+   ```
+
+2. Install the required Python packages:
+   ```bash
+   pip install -r requirements.txt
+   ```
+
+### Web Setup
+
+1. Navigate to the web directory:
+   ```bash
+   cd web
+   ```
+
+2. Install the required Node.js packages:
+   ```bash
+   npm ci
+   ```
+
+## Running the Application
+
+### Running the API (Backend)
+
+1. Navigate to the API directory:
+   ```bash
+   cd api
+   ```
+
+2. Start the FastAPI application:
+   ```bash
+   fastapi run main.py
+   ```
```

Comment:
This is awesome maybe add something about how the vscode settings allow for this to be run using F5. This also _may_ not work given that I usually put a virtual environment at the root of the project so it would need to be run from the root - can we fix that?